### PR TITLE
[FLINK-4590] [table] Some Table API tests are failing when debug lvl is set to DEBUG

### PR DESCRIPTION
**Pull request by twalthr**
Thanks for contributing to Apache Flink. Before you open your pull request, please take the following check list into consideration.
If your changes take all of the items into account, feel free to open your pull request. For more information and/or questions please refer to the [How To Contribute guide](http://flink.apache.org/how-to-contribute.html).
In addition to going through the list, please provide a meaningful description of your changes.

- [x] General
  - The pull request references the related JIRA issue ("[FLINK-XXX] Jira title text")
  - The pull request addresses only one issue
  - Each commit in the PR has a meaningful commit message (including the JIRA id)

- [x] Documentation
  - Documentation has been added for new functionality
  - Old documentation affected by the pull request has been updated
  - JavaDoc for public methods has been added

- [x] Tests & Build
  - Functionality added by the pull request is covered by tests
  - `mvn clean verify` has been executed successfully locally or a Travis build has passed

This PR fixes the estimated costs for minus and union DataSet RelNodes. I went through all RelNodes and also fixed multiple potential null pointers, bugs, and warnings.

### CODE REVIEW

File: flink-libraries/flink-table/src/test/resources/log4j-test.properties [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+# limitations under the License.
+################################################################################
+
+# Set root logger level to OFF to not flood build logs
+# set manually to INFO for debugging purposes
+log4j.rootLogger=OFF, testlogger
```
Comment (fhueske) [inline flink-libraries/flink-table/src/test/resources/log4j-test.properties]: I assume this is the actual fix for the referenced bug, right?

Comment (twalthr) [inline flink-libraries/flink-table/src/test/resources/log4j-test.properties]: As I explained in my PR description: "This PR fixes the estimated costs for minus and union DataSet RelNodes". This was the root cause for the failing Calcite assumptions. I just added the log4j file to easily enable assertions in future.

Comment (fhueske) [inline flink-libraries/flink-table/src/test/resources/log4j-test.properties]: Ah OK. The PR description didn't say that the missing cost functions were causing the failing tests.

**Comment by fhueske** [comment]
+1 to merge

**Pull request closed without merging**